### PR TITLE
Add Tax Exempt to Subscription

### PR DESCRIPTION
**Pull request by narefyev91**
### Details
Add Tax Exempt request flow to NewDot

Current PR is not covering middle stage - when Concierge is approving pdf. To check that flag is setting correctly in onyx
just add 
`                    Onyx.merge(ONYXKEYS.NVP_PRIVATE_TAX_EXEMPT, true);
`
here (for testing purpose)
Screenshot 2024-07-26 at 16 32 11

### Fixed Issues

$ https://github.com/Expensify/App/issues/45317
PROPOSAL:

### Tests
1. Open new.expensify.com/settings/subscriptions
2. Click "Request tax exempt status" button
3. Land in Concierge DM & Receive "I have received your request to set your account as tax exempt. Please attach a PDF of either your 501(c) or ST-119 here. Once that has been received and accepted, I'll get back to you here." message
4. Provide 501(c) or ST-119 proof, send to Concierge
5. Concierge marks account as tax exempt in Supportal
6. Verify that a tax exempt badge in new.expensify.com/settings/subscriptions

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open new.expensify.com/settings/subscriptions
2. Click "Request tax exempt status" button
3. Land in Concierge DM

### QA Steps
1. Open new.expensify.com/settings/subscriptions
2. Click "Request tax exempt status" button
3. Land in Concierge DM & Receive "I have received your request to set your account as tax exempt. Please attach a PDF of either your 501(c) or ST-119 here. Once that has been received and accepted, I'll get back to you here." message
4. Provide 501(c) or ST-119 proof, send to Concierge
5. Concierge marks account as tax exempt in Supportal
6. Verify that a tax exempt badge in new.expensify.com/settings/subscriptions

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x  I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot_1722337076

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b0edf794-7847-495a-af07-8405dac9bd06

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screenshot - iPhone 15 Plus - 2024-07-30 at 13 45 44

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/e2c097e8-2a57-408a-b1ad-0e0706a321a1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/71520436-cfe4-4ce2-9a01-f6e00c7ed5ba

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/b706ffab-91cf-4b43-aacb-474f853ce0af

</details>

### CODE REVIEW

**Comment by narefyev91** [comment]
@Expensify/design  can i get a review for UI please
Screenshot 2024-07-25 at 15 17 20
Simulator Screenshot - iPhone 15 Pro - 2024-07-25 at 15 18 59
Screenshot 2024-07-25 at 19 00 30

**Comment by shawnborton** [comment]
I think that looks pretty good to me. Does the overflow icon stay "highlighted" once the popover is active, like this example in Wallet > Bank accounts?
CleanShot 2024-07-25 at 18 07 03@2x

And I assume the badge will only show up once we get the flag from the BE that the account is tax-exempt?

**Comment by narefyev91** [comment]
> I think that looks pretty good to me. Does the overflow icon stay "highlighted" once the popover is active, like this example in Wallet > Bank accounts? CleanShot 2024-07-25 at 18 07 03@2x
> 
> And I assume the badge will only show up once we get the flag from the BE that the account is tax-exempt?

yeah - using standard component which we already have
Screenshot 2024-07-25 at 19 13 22

**Comment by shawnborton** [comment]
Yeah sorry, my question is that in some places of the app, when you press the overflow icon, it stays green while the menu is open (see my screenshot). But maybe we don't do that everywhere. Any ideas?

**Comment by dannymcclain** [comment]
Looking good to me too. I agree with Shawn about the three-dot icon being highlighted when the menu is open. Not critically necessary, but if we can make it match that Wallet example I think that'd be nice. But yeah, as Shawn pointed out, I'm not sure that behavior is totally consistent throughout the app (it doesn't happen in the workspace list for example).

Also, should we remove the three-dot menu once they have tax-exempt status? Or do we need to leave it and change the menu item? (No idea what it would be if they've already requested tax-exemption.)

**Comment by shawnborton** [comment]
That's a great point... @trjExpensify any thoughts there? When the tax exempt status is in place, do we just get rid of the overflow or change the option to be something like "Remove tax exempt status"? No idea if anyone would ever want/need to do that though.

**Comment by narefyev91** [comment]
> Looking good to me too. I agree with Shawn about the three-dot icon being highlighted when the menu is open. Not critically necessary, but if we can make it match that Wallet example I think that'd be nice. But yeah, as Shawn pointed out, I'm not sure that behavior is totally consistent throughout the app (it doesn't happen in the workspace list for example).
> 
> Also, should we remove the three-dot menu once they have tax-exempt status? Or do we need to leave it and change the menu item? (No idea what it would be if they've already requested tax-exemption.)

Yeah when tax-exempt exists we will not shown 3 dots - it's just screenshot to see all UI elements at once :-)

**Comment by narefyev91** [comment]
@shawnborton @dannymcclain do we want to make 3 dots being highlighted globally? Just to make it the same for all pages?
If we already have some inconsistency between pages - does it make sense to unified?

**Comment by shawnborton** [comment]
I think we should unify the behavior globally, yes. Do you know why it's currently inconsistent though?

**Comment by narefyev91** [comment]
> I think we should unify the behavior globally, yes. Do you know why it's currently inconsistent though?

Probably we do not implement logic to switch color - and as i see for Wallet page - it's a custom component not global <ThreeDotsMenu />. I can just add logic for changing color of icon if popover is open in one place - and all pages will reflect that

**Comment by dannymcclain** [comment]
Awesome. I think that'd be great.

**Comment by narefyev91** [comment]
> Awesome. I think that'd be great.

Cool - will add as a first thing tomorrow - and will add screenshots

**Comment by narefyev91** [comment]
@shawnborton @dannymcclain here we go:
Screenshot 2024-07-26 at 11 41 41

**Comment by shawnborton** [comment]
Looks good to me!

**Comment by dannymcclain** [comment]
>Looks good to me!

Same! 🚀

**Comment by narefyev91** [comment]
@hoangzinh fixed both bugs above!

**Comment by shawnborton** [comment]
Can you share an updated screenshot please? Thanks!

**Comment by shawnborton** [comment]
Nice, that looks good to me.

File: src/libs/API/types.ts [hunk 1]
```diff
@@ -318,6 +318,7 @@ const WRITE_COMMANDS = {
     UPDATE_SAGE_INTACCT_NON_REIMBURSABLE_EXPENSES_CREDIT_CARD_CHARGE_EXPORT_DEFAULT_VENDOR: 'UpdateSageIntacctNonreimbursableExpensesCreditCardChargeExportDefaultVendor',
     UPDATE_SAGE_INTACCT_NON_REIMBURSABLE_EXPENSES_EXPORT_ACCOUNT: 'UpdateSageIntacctNonreimbursableExpensesExportAccount',
     UPDATE_SAGE_INTACCT_NON_REIMBURSABLE_EXPENSES_EXPORT_VENDOR: 'UpdateSageIntacctNonreimbursableExpensesExportVendor',
+    REQUEST_SUBSCRIPTIONS_TAX_EXEMPT: 'RequestTaxExemption',
```
Comment (marcaaron) [inline src/libs/API/types.ts:321]: Can we keep this consistent with the current command name? So just `REQUEST_TAX_EXEMPTION`?

File: src/libs/actions/Subscription.ts [hunk 2]
```diff
@@ -288,6 +288,10 @@ function cancelBillingSubscription(cancellationReason: FeedbackSurveyOptionID, c
     API.write(WRITE_COMMANDS.CANCEL_BILLING_SUBSCRIPTION, parameters);
 }
 
+function requestSubscriptionTaxExempt() {
```
Comment (marcaaron) [inline src/libs/actions/Subscription.ts:291]: I'd think it's ok if this is just `requestTaxExempt()` as well.

**Review by marcaaron — COMMENTED** [review]
Looking great so far. Had a small nit, but don't see any major blockers. 🚀

**Comment by narefyev91** [comment]
> @narefyev91 can you add screenshots for native android/ios as well? Just to ensure we don't break existing UI on those native platforms. Also, can you check again "Offline tests"? There are some steps that we can't do when offline.

Probably for offline - we just can only lands of concierge chat and waiting until user gets online. (updated)
Added IOS screenshot, waiting for android to be build

**Review by hoangzinh — APPROVED** [review]
LGTM

**Review by marcaaron — APPROVED** [review]
LGTM 🎉

**Comment by kavimuru** [comment]
@marcaaron @hoangzinh How should we run the following?

> Land in Concierge DM & Receive "I have received your request to set your account as tax exempt. Please attach a PDF of either your 501(c) or ST-119 here. Once that has been received and accepted, I'll get back to you here." message
> Provide 501(c) or ST-119 proof, send to Concierge
> Concierge marks account as tax exempt in Supportal

**Comment by kavimuru** [comment]
@hoangzinh what about step 4 Provide 501(c) or ST-119 proof, send to Concierge

**Comment by marcaaron** [comment]
This requires supportal access - but I just tested it on staging and we can check it off.

**Pull request merged**